--- CA.pl.orig	2020-07-01 03:34:43.171025469 -0300
+++ CA.pl.new	2020-07-01 03:40:25.328197984 -0300
@@ -27,19 +27,19 @@
 my $OPENSSL_CONFIG = $ENV{"OPENSSL_CONFIG"} || "";
 my $DAYS = "-days 365";
 my $CADAYS = "-days 1095";	# 3 years
-my $REQ = "$openssl req $OPENSSL_CONFIG";
-my $CA = "$openssl ca $OPENSSL_CONFIG";
+my $REQ = "$openssl req -config $OPENSSL_CONFIG";
+my $CA = "$openssl ca -config $OPENSSL_CONFIG";
 my $VERIFY = "$openssl verify";
 my $X509 = "$openssl x509";
 my $PKCS12 = "$openssl pkcs12";
 
 # default openssl.cnf file has setup as per the following
-my $CATOP = "./demoCA";
+my $CATOP = $ENV{"OPENSSL_CATOP"} || "./demoCA";
 my $CAKEY = "cakey.pem";
 my $CAREQ = "careq.pem";
 my $CACERT = "cacert.pem";
 my $CACRL = "crl.pem";
-my $DIRMODE = 0777;
+my $DIRMODE = 0770;
 
 my $NEWKEY = "newkey.pem";
 my $NEWREQ = "newreq.pem";
